client/src/NewEvent.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import { EventsApi, EventInput } from "../api-client"; // import klienta a typů
+
+const NewEvent: React.FC = () => {
+    const [title, setTitle] = useState("");
+    const [location, setLocation] = useState("");
+    const [dates, setDates] = useState<string[]>([""]);
+
+    const handleSubmit = async (e: React.FormEvent) => {
+        e.preventDefault();
+
+        const payload: EventInput = {
+            title,
+            location,
+            dates: dates
+                .map((d) => new Date(d))
+                .filter((d) => !isNaN(d.getTime()))
+                .map((d) => ({ timestamp: d.getTime(), records: [] })),
+        };
+
+        try {
+            const api = new EventsApi();
```

Comment:
tady musis mit toho api clienta, ale ja ho nikde nevidim. Je treba
```
npm install openapi-fetch
```

a pak to vyuzit pro napsani metod, co fetchuji data nebo posilaji formular, napr.

```ts
import createClient from "openapi-fetch";
import type { paths, components } from "./api-types"; // generated by openapi-typescript
export type SimplePollingEvent = components["schemas"]["Event"];

const client = createClient<paths>({ baseUrl: "http://localhost:4000" });

export const loadEvents = async () => {
  const { data, error } = await client.GET("/api/events");
  if (error) {
    throw new Error(error);
  }

  return data.items;
};

```

a je to :)